sequencer/src/bin/nasty-client.rs

```diff
@@ -940,11 +936,13 @@ impl ResourceManager<BlockQueryData<SeqTypes>> {
                     .context(format!("fetching header {block}"))
             })
             .await?;
-        if header.ns_table.is_empty() {
+        let num_namespaces = header.ns_table.iter().count();
+        if num_namespaces == 0 {
             tracing::info!("not fetching namespace because block {block} is empty");
```

Comment:
If `num_namespaces == 0` then it's impossible even to create a `NsIndex` needed to retrieve a `NamespaceId` for the query, thanks to my militant use of newtypes.

In order to execute the query I'd need to guess a namespace ID. The only thing I could test with such a query is that my guessed ID caused the endpoint to return an error, but I expect that outcome anyway even for a non-empty block because the ID was guessed.

Current `main` branch does the same thing: just return `Ok(())` immediately if the namespace table is empty. So I'm going to leave this as-is.